app/widgets/common/inputs/fields/useFetchUrlWithDeploymentId.ts

```diff
@@ -1,20 +1,36 @@
-import { useContext, useMemo } from 'react';
+import { useCallback, useContext, useMemo } from 'react';
 import DeploymentIdContext from '../utils/deploymentIdContext';
+import BlueprintIdContext from '../utils/blueprintIdContext';
 import getConstraintValueFunction from '../utils/getConstraintValueFunction';
 import type { Constraint } from '../types';
 
-export default function useFetchUrlWithDeploymentId(fetchUrl: string, constraints: Constraint[]) {
+export default function useFetchUrlWithDeploymentId(
+    fetchUrl: string,
+    constraints: Constraint[],
+    preferBlueprintId?: boolean
```

Comment:
To avoid boolean parameter (as boolean parameters are often considered to be a code smell) maybe we could try IMHO a bit more elegant solution - options object parameter, so that `preferBlueprintId` is clearly visible when calling this hook?

```suggestion
    { constraints: Constraint[], preferBlueprintId?: boolean }
```

On the other hand - I'm not sure we actually need that additional parameter. As we are always getting blueprint ID from the context anyway (L14), why not try to use it always? 🤔

Reply:
Applied as a part of 872ac62b661f3ac01a14f60146d5cd33335c5f02